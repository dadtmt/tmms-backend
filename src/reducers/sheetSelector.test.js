import reducer from './sheetSelector'

const initialState = {
  selectedSheet: {
    characteristics: [],
    inventory: [],
    name: 'type a name for this character'
  }
}

it('return initial state', () => {
  expect(reducer(
    initialState,
    {
      type: 'SOME_ACTION_TYPE'
    }
  )).toMatchSnapshot()
})

it('handles SELECT_SHEET', () => {
  expect(reducer(
    initialState,
    {
      sheet: {
        id: 'SOME_ID',
        name: 'Marom'
      },
      type: 'SELECT_SHEET'
    }
  )).toMatchSnapshot()
})

it('handles CREATE_SHEET_MUTATION', () => {
  expect(reducer(
    initialState,
    {
      operationName: 'CreateSheet',
      result: {
        data: {
          createSheets: {
            changedEdge: {
              node: {
                description: 'semi orc',
                id: 'SOME_ID',
                image: 'marom.png',
                isPlayer: true,
                name: 'Marom'
              }
            }
          }
        }
      },
      type: 'APOLLO_MUTATION_RESULT'
    }
  )).toMatchSnapshot()
})
