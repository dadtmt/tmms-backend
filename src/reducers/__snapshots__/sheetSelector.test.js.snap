exports[`test handles CREATE_SHEET_MUTATION 1`] = `
Object {
  "selectedSheet": Object {
    "description": "semi orc",
    "id": "SOME_ID",
    "image": "marom.png",
    "isPlayer": true,
    "name": "Marom",
  },
}
`;

exports[`test handles SELECT_SHEET 1`] = `
Object {
  "selectedSheet": Object {
    "id": "SOME_ID",
    "name": "Marom",
  },
}
`;

exports[`test return initial state 1`] = `
Object {
  "selectedSheet": Object {
    "characteristics": Array [],
    "inventory": Array [],
    "name": "type a name for this character",
  },
}
`;
